cmd/remote-work-processor/main.go

```diff
@@ -17,80 +17,160 @@ limitations under the License.
 
 import (
-	// "flag"
-	// "os"
-
-	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
-	// to ensure that exec-entrypoint and run can make use of them.
-
+	"context"
+	"crypto/sha256"
+	_ "embed"
+	"encoding/hex"
+	"flag"
+	"fmt"
+	"github.com/SAP/remote-work-processor/internal/grpc"
+	"github.com/SAP/remote-work-processor/internal/grpc/processors"
+	"github.com/SAP/remote-work-processor/internal/kubernetes/controller"
+	meta "github.com/SAP/remote-work-processor/internal/kubernetes/metadata"
+	"io"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
 	"log"
 	"os"
-
+	"os/signal"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"strconv"
+	"syscall"
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	// to ensure that exec-entrypoint and run can make use of them.
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
-
-	"k8s.io/apimachinery/pkg/runtime"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
-
-	// "sigs.k8s.io/controller-runtime/pkg/healthz"
-	// "sigs.k8s.io/controller-runtime/pkg/log/zap"
-	// "github.com/SAP/remote-work-processor/kubernetes/controllers"
-	"github.com/SAP/remote-work-processor/internal/grpc"
-	"github.com/SAP/remote-work-processor/internal/grpc/processors"
-	"github.com/SAP/remote-work-processor/internal/kubernetes/controller"
-	"github.com/SAP/remote-work-processor/internal/kubernetes/metadata"
 	//+kubebuilder:scaffold:imports
 )
 
-var (
-	scheme = runtime.NewScheme()
+const (
+	standaloneModeOpt = "standalone-mode"
+	instanceIdOpt     = "instance-id"
+	connRetriesOpt    = "connection-retries"
 )
 
-func init() {
-	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
-	//+kubebuilder:scaffold:scheme
-}
+var Version string
 
 func main() {
-	metadata.InitRemoteWorkProcessorMetadata()
-	config := getKubeConfig()
+	setupFlagsAndLogger()
 
-	e := controller.CreateManagerEngine(scheme, config)
-	processors.InitProcessorFactory(e)
-	grpc.InitRemoteWorkProcessorGrpcClient()
+	rootCtx, _ := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
 
-	opc := grpc.Client.Receive()
+	instanceIDFlag := flag.Lookup(instanceIdOpt).Value.String()
+	rwpMetadata := meta.LoadMetadata(instanceIDFlag, Version)
+	grpcClient := grpc.NewClient(rwpMetadata)
+	drainChan := make(chan struct{}, 1)
 
-	for {
-		op := <-opc
-		p, err := processors.Factory.CreateProcessor(op)
-		if err != nil {
-			log.Fatalf("Error occurred while creating operation processor: %v\n", err)
-		}
+	var factory processors.ProcessorFactory
 
-		res := <-p.Process()
-		if res.Err != nil {
-			log.Fatalf("Error occurred while processing operation: %v\n", err)
-		}
+	isInStandaloneMode, _ := strconv.ParseBool(flag.Lookup(standaloneModeOpt).Value.String())
```

Comment:
The `flag` package handles parsing errors on the flags. If something that is not a boolean is provided for `--standalone-mode` it will fail with a parse error.